boot/zephyr/serial_adapter.c

```diff
@@ -194,6 +194,13 @@ boot_uart_fifo_init(void)
 	uart_dev = device_get_binding(CONFIG_RECOVERY_UART_DEV_NAME);
 #elif CONFIG_BOOT_SERIAL_CDC_ACM
 	uart_dev = device_get_binding(CONFIG_USB_CDC_ACM_DEVICE_NAME "_0");
+	if (uart_dev) {
+		int rc;
+		rc = usb_enable(NULL);
+		if (rc) {
+			return (-1);
+		}
+	}
```

Comment:
Why not like this? Imo its cleaner

```diff
 #elif CONFIG_BOOT_SERIAL_CDC_ACM
        uart_dev = device_get_binding(CONFIG_USB_CDC_ACM_DEVICE_NAME "_0");
-       if (uart_dev) {
-               int rc;
-               rc = usb_enable(NULL);
-               if (rc) {
-                       return (-1);
-               }
-       }
 #endif
        u8_t c;
 
        if (!uart_dev) {
                return (-1);
        }
+
+       int rc;
+       rc = usb_enable(NULL);
+       if (rc) {
+               return (-1);
+       }
+

```

Reply:
usb_enable will fail outside of #ifdery when compiling for serial-uart recovery.

Reply:
That's right of curse, sorry.